go/vt/vtgate/executor.go

```diff
@@ -698,6 +740,11 @@ func (e *Executor) handleShow(ctx context.Context, safeSession *SafeSession, sql
 	defer func() { logStats.ExecuteTime = time.Since(execStart) }()
 
 	switch show.Type {
+	case sqlparser.KeywordString(sqlparser.TABLES):
+		if show.ShowTablesOpt != nil && show.ShowTablesOpt.DbName != "" {
+			show.ShowTablesOpt.DbName = "vt_" + destKeyspace
```

Comment:
The db name can be overridden in the vttablet command line. So, this won't work in all cases.
I think a better solution will be to lookup the keyspace for the db name, and issue a simple `show tables` (without the `from`) to a vttablet of that keyspace. This will also be consistent from the fact that we intercept `show databases` to return the list of keyspaces.